spec/formatting.md

```diff
@@ -222,6 +222,61 @@ the following steps are taken:
 
    The form that resolved _operand_ and _option_ values take is implementation-defined.
 
+   A _local-declaration_ binds the resolved value of an _expression_
+   to a _variable_.
+   Thus, the output of one _function_ is potentially the _operand_
+   of another _function_. In other words, formatting functions
+   compose with each other.
+   For example, in
+   ```
+   .input {$n :number minIntegerDigits=3}
+   .local {$n1 :number maxFractionDigits=3}
+   ```
+   the second call to `:number` composes with the first call.
```

Comment:
I think adding the idea of "compose" is going too far. We should be very conservative here. Also, this still has the "output" of a function in play. Perhaps:

```suggestion
Thus, the _operand_ for one _function_ might be the resolved value
of another _function_.
Further, the _options_ for one _expression_ might affect the operation
of another.

>For example, if the value of the variable `n` were `1`:
> ```
> .input {$n :number minimumFractionDigits=1}
> .local $num = {$n :number minimumIntegerDigits=3}
> .match {$num}
> *   {{Prints 001.0 for {$num}}}
> ```
> ... because the _options_ for the `.input` and `.local` are both applied to the value
> for the purposes of both formatting and selection.
> (Note that in English, fractional values match the plural rule `other`)
```

Also note that the `.local` in the original is incorrect.

Reply:
Fixed the indents in the above suggestion. Its example should be updated, because we should not be showing any matching (even if only with a single `*` variant) on a non-integral number. Then could also drop the parenthetical and irrelevant bit about `other`.

Reply:
So the example could read:

```
.input {$n :number minimumFractionDigits=1}
.local $num = {$n :number minimumIntegerDigits=3}
{{Prints 001.0 for {$num}}}
```

Reply:
I don't mind using the word "compose" here, but I'm also OK with dropping it and replacing with a description of the observed behavior, like @aphillips proposed.

I would, however, edit @aphillips's suggestion slightly:

```diff
-Thus, the _operand_ for one _function_ might be the resolved value
-of another _function_.
+Thus, the resolved value of one _function_ might be the _operand_
+or an _option_ value for one another _function_.
 Further, the _options_ for one _expression_ might affect the operation of another.
```

This does two things:

* The subject of the sentence is the resolved value, similar to the second sentence.
* It's not only operands that can be resolved values of other expressions; option values can as well, as per our syntax and data model.

Reply:
>  because we should not be showing any matching (even if only with a single * variant) on a non-integral number.

`:number` really does need to do plural matching on fractions and this isn't a problem. Your concern is, I think, about exact matching, which I do not show. I actually think the `.match` is important to call out precisely because the fraction bits happen to `$n`

Reply:
Ah, fair point, I was confused. Including a `one {{This is never selected}}` would clarify things a bit.

Reply:
LOL. Actually, I had it originally, but removed it before submitting--I thought for clarity!

Reply:
"the resolved value   of another _function_" doesn't really make sense here. The resolved value of a _function_ is a thing that represents the function; the resolved value of a function applied to arguments (using more conventional declarative-language terminology) is something left unspecified, but is _not_ a representation of a function itself. The spec doesn't really offer us the language for disambiguating the two.

Reply:
I'm not sure if replacing the original example with the one with `.match` clarifies the point, although more examples are generally a good thing. I wanted to show an example with two formatting functions, because that's where it might not be obvious that functions compose.